bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableTableAdminClient.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.core;
+
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.admin.v2.models.ModifyColumnFamiliesRequest;
+import com.google.cloud.bigtable.admin.v2.models.Table;
+import com.google.common.util.concurrent.ListenableFuture;
+import java.util.List;
+
+/**
+ * Interface to access Bigtable gax table admin API.
+ */
+public interface IBigtableTableAdminClient {
+
+  /**
+   * Creates a new table. The table can be created with a full set of initial column families,
+   * specified in the request.
+   * @param request a {@link CreateTableRequest} object.
+   */
+  Table createTable(CreateTableRequest request);
+
+  /**
+   * Creates a new table asynchronously. The table can be created with a full set of initial column
+   * families, specified in the request.
+   *
+   * @param request a {@link CreateTableRequest} object.
+   */
+  //TODO(rahulkql): Once it is adapted to v2.models, change the return type to ApiFuture.
+  ListenableFuture<Table> createTableAsync(CreateTableRequest request);
```

Comment:
I'm confused about the api here. The parameter is a model from google-cloud-java, but the return type is a raw proto.

Reply:
the purpose for this interface is to adapt admin client calls to google-cloud-java models. As of now we are not directly using GCJ's `BigtableDataClient` & `BigtableTableAdminClient`( as client conversion is still pending in #1973). 

Also the return type is GCJ's `com.google.cloud.bigtable.admin.v2.models.Table`

Reply:
My apologies, I misread